### ISSUE TITLE
VoxelgridSampler broken when using with single_fit

### ISSUE THREAD

**Issue by cosama** [daavoo/pyntcloud#149]
Trying to use single_fit with the VoxelgridSampler outputs the following error message:

>~/.local/lib/python3.6/site-packages/pyntcloud-0.1.0-py3.6.egg/pyntcloud/ransac/samplers.py in >get_sample(self)
>     67         # pick one point and get it's voxel index
>     68         idx = np.random.randint(0, len(self.points))
>---> 69         voxel = self.voxelgrid.voxel_n[idx]
>     70 
>     71         # get index of points inside that voxel and convert to probabilities
>
>AttributeError: 'VoxelGrid' object has no attribute 'voxel_n'

Code that causes the error:

`is_plane, model = single_fit(cloud.xyz, RansacPlane, VoxelgridSampler, return_model=True,
                             model_kwargs={"max_dist": 0.2}, max_iterations=100,
                             sampler_kwargs={"voxel_size": 0.1})`

I assume this is because it never calls the compute() member function, however I'm not sure how to fix it properly.

**Comment by daavoo** [comment]
Thanks for pointing out this bug. As you pointed, self.voxelgrid.compute is not being called in VoxelgridSampler.__init__

**Comment by daavoo** [comment]
Hola @cosama!. It took me a while to found time to fix this (#188) and I ended up refactoring a lot of sutff to make the structures more usable from outside the PyntCloud class.

In theory (works for my and for some simple test), you can run:

```python
is_plane, model = single_fit(
    points=cloud.xyz,
    model=RansacPlane,
    model_kwargs={"max_dist": 0.2},
    return_model=True,
    sampler=VoxelgridRansacSampler,
    sampler_kwargs={"size_x": 0.1, "size_y": 0.1, "size_z": 0.1},
    max_iterations=100
)
```

And the result should be what you were trying to do when you faced the bug.
